src/main/scala/org/constellation/consensus/EdgeProcessor.scala

```diff
@@ -478,6 +478,11 @@ object Snapshot {
 
       // TODO: CheckpointService - apply snapshot (remove from mem pool, add to midDb)
 
+      dao.checkpointService.memPool.remove(cb.baseHash).unsafeRunSync()
+      dao.checkpointService.midDb.remove(cb.baseHash).unsafeRunSync()
+      dao.transactionService.merklePool.remove(cb.transactionsMerkleRoot).unsafeRunSync()
```

Comment:
we shouldn't remove directly from internals of transaction service. there is a method `applySnapshot` and imo there we should do that /\

Reply:
Updated